test/dao.suite/patch-by-id.suite.js

```diff
@@ -0,0 +1,188 @@
+            done();
+          });
+        });
+    });
+
+    it('should throw error on unknown attributes when strict: throw', function(done) {
+      Person.definition.settings.strict = 'throw';
+      Person.findById(personId, function(err, p) {
+        Person.patchById(personId, { foo: 'bar' },
+          function(err, info) {
+            should.exist(err);
+            err.name.should.equal('Error');
+            err.message.should.equal('Unknown property: foo');
+            should.not.exist(info);
+            Person.findById(personId, function(e, p) {
+              if (e) return done(e);
+              should.not.exist(p.foo);
+              done();
+            });
+          });
+      });
+    });
+
+    it('should throw error on unknown attributes when strict: validate', function(done) {
+      Person.definition.settings.strict = 'validate';
+      Person.findById(personId, function(err, p) {
+        Person.patchById(personId, { foo: 'bar' },
+          function(err, info) {
+            should.exist(err);
+            err.name.should.equal('ValidationError');
+            err.message.should.containEql('`foo` is not defined in the model');
+            Person.findById(personId, function(e, p) {
+              if (e) return done(e);
+              should.not.exist(p.foo);
+              done();
+            });
+          });
+      });
+    });
+
+    it('should allow same id value on patchById', function(done) {
+      Person.patchById(personId, { id: personId, name: 'John' },
+        function(err, info) {
+          if (err) return done(err);
+          Person.findById(personId, function(e, p) {
+            if (e) return done(e);
+            p.name.should.equal('John');
+            p.age.should.equal(15);
+            done();
+          });
+        });
+    });
+
+    it('should allow same stringified id value on patchById',
+      function(done) {
+        var pid = personId;
+        if (typeof pid === 'object' || typeof pid === 'number') {
+          // For example MongoDB ObjectId
+          pid = pid.toString();
+        }
+        Person.patchById(pid, { id: pid, name: 'John' },
+          function(err, info) {
+            if (err) return done(err);
+            Person.findById(personId, function(e, p) {
+              if (e) return done(e);
+              p.name.should.equal('John');
+              p.age.should.equal(15);
+              done();
+            });
+          });
+      });
+
+    it('should fail if an id value is to be changed on patchById',
+      function(done) {
+        Person.patchById(personId, { id: personId + 1, name: 'John' },
+        function(err, info) {
+          should.exist(err);
+          done();
+        });
+      });
+
+    it.skip('should allow model instance on patchById', function(done) {
+      // QUESTION-test[1]: Can we pass undefined here and should we remove undefined in the method?
+      // Please see the other comment in patch-by-id.js
```

Comment:
Yeah, it is a know problem that you cannot pass `age:undefined` to clear a value via updateAttributes. I think this is also a limitation of MongoDB. That's one of the reasons for introducing `replaceById`.